src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -58,48 +59,56 @@ export const useTxActions = (): TxActions => {
 
     const addToBatch: TxActions['addToBatch'] = async (safeTx, origin) => {
       assertTx(safeTx)
-      assertWallet(wallet)
+      assertProvider(signer?.provider)
 
-      const tx = await proposeTx(wallet.address, safeTx, undefined, origin)
+      const tx = await proposeTx(signer.address, safeTx, undefined, origin)
       await addTxToBatch(tx)
       return tx.txId
     }
 
     const signRelayedTx = async (safeTx: SafeTransaction, txId?: string): Promise<SafeTransaction> => {
       assertTx(safeTx)
-      assertWallet(wallet)
+      assertProvider(signer?.provider)
 
       // Smart contracts cannot sign transactions off-chain
-      if (await isSmartContractWallet(wallet.chainId, wallet.address)) {
+      if (await isSmartContractWallet(signer.chainId, signer.address)) {
         throw new Error('Cannot relay an unsigned transaction from a smart contract wallet')
       }
-      return await dispatchTxSigning(safeTx, version, wallet.provider, txId)
+      return await dispatchTxSigning(safeTx, version, signer.provider, txId)
     }
 
     const signTx: TxActions['signTx'] = async (safeTx, txId, origin) => {
       assertTx(safeTx)
-      assertWallet(wallet)
+      assertProvider(signer?.provider)
       assertOnboard(onboard)
 
       // Smart contract wallets must sign via an on-chain tx
-      if (await isSmartContractWallet(wallet.chainId, wallet.address)) {
+      if (await isSmartContractWallet(signer.chainId, signer.address)) {
```

Comment:
We could add the `signer.isSafe` flag here as well meaning Safes don't require the `isSmartContractWallet` call.